src/daemon/daemon.cpp

```diff
@@ -2040,6 +2059,22 @@ void mp::Daemon::create_vm(const CreateRequest* request, grpc::ServerWriter<Crea
 {
     auto checked_args = validate_create_arguments(request);
 
+    // Validate the names of the networks to connect to. Check here, because this function is inside the daemon,
```

Comment:
There is a call to `validate_create_arguments` just above, so I don't think it's appropriate to `validate_network_ids` just after. If you need the factory to check the ids, you can pass it on as a parameter. 

`validate_create_arguments` in turn calls `validate_extra_interfaces`, which is where the check should be. That would keep it tidy and free another 15 LOC in this already big function. The factory would just need to be forwarded there. 

(Alternatively, you could make those helpers private methods, but I don't think that's necessary for this.)

Reply:
Ouch, right. I could save a lot of spreading by just passing a parameter. I did pass a `factory_networks` parameter to `validate_create_arguments`, which is itself passed to `validate_extra_interfaces` and everything is checked in the latter function.